syntax = "proto3";

service TestCaseService {
  rpc TestFields (FieldsMessage) returns (FieldsMessage) {}
  rpc GetAccount (GetAccountRequest) returns (Account) {}
}

message FieldsMessage {
  int32 int32_field = 1;
  float float_field = 2;
  double double_field = 3;
  bool bool_field = 4;
  string string_field = 5;
  bytes bytes_field = 6;
  repeated string repeated_string_field = 7;
  Account nested_field = 8;
  repeated Account repeated_nested_field = 9;

}

message GetAccountRequest {
  string account_id = 1;
}

message Account {
  string username = 1;
  int32 balance = 2;
  AccountSettings settings = 3;
}

message AccountSettings {
  bool is_public = 1;
}